src/compose/network.ts

```diff
@@ -28,50 +23,45 @@ export class Network {
 
 	private constructor() {}
 
-	public static fromDockerNetwork(network: NetworkInspect): Network {
+	public static fromDockerNetwork(
+		network: dockerode.NetworkInspectInfo,
+	): Network {
 		const ret = new Network();
 
 		const match = network.Name.match(/^([0-9]+)_(.+)$/);
 		if (match == null) {
 			throw new InvalidNetworkNameError(network.Name);
 		}
+
+		// TODO: this seems uncessary with the regex above
```

Comment:
Agreed. Looking at the regex, the `|| null` bit seems unnecessary. None of the following examples match the regex except for the all-numbers one:
![image](https://user-images.githubusercontent.com/38748558/116151104-29ad9a80-a71f-11eb-8bf6-9f63c9f9db89.png)

The `checkInt` call might be required to convert string to int though, since the regex would match and return a string from the network string. checkInt will return `undefined` for non-integer parameters, so the `|| null` is doubly unnecessary.

Reply:
So if, the network name doesn't match the regex, then the name doesn't contain an appId, and the first `if` will catch it. Even doing `checkInt` is unnecessary because the first part of the match should be a number at that point

Reply:
`checkInt` converts the value into a number type which the regex test won't btw.

Reply:
I'd leave it since we only gain 4 LOC